### PR TITLE
SDIT-1955 Create a new index to track the green/blue indices

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt [hunk 1]
```diff
@@ -469,3 +469,6 @@ class PrisonerA : Prisoner()
 
 @Document(indexName = "prisoner-search-b")
 class PrisonerB : Prisoner()
+
+@Document(indexName = "prisoner-search")
+class PrisonerC : Prisoner()
```
Comment (steverendell) [inline common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt:474]: Not sure why the existing index names dont match the SyncIndex index names but this one does

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/PrisonerSynchroniserService.kt [hunk 2]
```diff
@@ -74,6 +77,7 @@ class PrisonerSynchroniserService(
       prisonerDifferenceService.reportDiffTelemetry(existingPrisoner, prisoner)
 
       indices.map { prisonerRepository.save(prisoner, it) }
+      prisonerRepository.save(prisoner, SyncIndex.RED)
       prisonerDifferenceService.handleDifferences(existingPrisoner, ob, prisoner, "REFRESH")
     }
```
Comment (steverendell) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/PrisonerSynchroniserService.kt:82]: An index build will now also populate the new index

File: hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/resource/CompareIndexResourceIntTest.kt [hunk 3]
```diff
@@ -180,4 +185,105 @@ class CompareIndexResourceIntTest : IntegrationTestBase() {
...
+    inner class HappyPath {
+      private val pA9999RA = PrisonerBuilder("A9999RA")
+
+      @BeforeEach
+      fun beforeEach() {
+        prisonApi.stubOffenders(
+          PrisonerBuilder("A9999AA"),
+          PrisonerBuilder("A9999AB"),
+          PrisonerBuilder("A9999AC"),
+          pA9999RA,
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/resource/CompareIndexResourceIntTest.kt:224]: typo? surprised it compiles!

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt (hunk 1 above)
Comment (petergphillips) [reply common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt:474]: don't think these are used though - think `-a` and `-b` can be removed instead

**Review by andymarke — DISMISSED** [review]
lgtm

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt (hunk 1 above)
Comment (steverendell) [reply common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/Prisoner.kt]: Great, have removed.

**Review by andymarke — APPROVED** [review]

**Pull request merged**